tests/test_rooms.py

```diff
@@ -602,3 +602,28 @@ def test_pinning(room, room2, user, mod, admin, global_admin, no_rate_limit):
         room2.pin(7, global_admin)
 
     assert not room2.pinned_messages
+
+
+def test_active_users(room, user, user2):
```

Comment:
i feel like the usage of cleanup and _refresh can be done with some kind of context manager here.

Reply:
Is it actually worth bothering, just for test code like this?

Reply:
no